### ISSUE TITLE
npme: "Using npm Enterprise" is totally unclear

### ISSUE THREAD

**Issue by evil-shrike** [npm/docs#88]
I've installed npme. Run it. Opened "http://localhost:8080/" in my browser and see some json. Nice. What next?

Looking at https://docs.npmjs.com/enterprise/using-it and see nothing that helps.

> Get the latest npm client

It's mentioned already on Installation page

Next
>Authenticating
>Once you have an up-to-date client, log in to your registry using your GitHub or GitHub Enterprise credentials

Mmm.. Why? As I understand `npm login` is needed to automatically pick right registry for a scope. But is npme only for scoped packages? I hope it's not.
First thing I'd try (and did) is to try to install a public package via my npmE:
`npm install -g grunt-cli --registry=http://localhost:8080"`
but.. it doesn't work ("Package not found : grunt-cli") ! 
So probably npmE doesn't proxy public npmjs.org by default. How to make it? Probably we need some configuration. Let's look into /etc/npme/ folder. Hurray, there's `service.json` file looking like a config file. 
Inside service.json I can see something looking like modules and options. Should it mentioned in the docs?
Module 'policy-follower' is probably what I need for replicating public npm. Looking in readme.md of \@npm/node-modules/policy-follower doesn't help at all. How to configure it?

Next topics "installing", "requiring" and "publishing" talk about only scoped packaged. Why? It looks like npme can be used only for scoped packaged.

So I'd put info in the following order:

1. how to install a public package (grunt-cli, bower, etc) via npme.
profit - caching, fault tolerance
Document: Specifiing --registry in `npm install`, setting the npmE to replicating all (or only some) packages.

2. how to put a package into npme
Profit: sharing a package among enterprise.
Recommend: use scoped packages as they can be easily installed from npme without specifing `--registry` - put a reference to https://docs.npmjs.com/misc/scope
Anyway to publish a package (scoped or not) we should authenticate ourselves:

* how to do it for client (probably `npm adduser` or `npm login`?)
* how to do it for server npme - how? Npme is configured to use Github. But obviously we wont allow all github users to publish packages into our private registry. So how to limit to particular users?

**Comment by owais** [comment]
@linclark The question following question is still unanswered. 

>  "how to do it for server npme - how? Npme is configured to use Github. But obviously we wont allow all github users to publish packages into our private registry. So how to limit to particular users?"

I've setup npme on my private server but how do I determine which users should be able to login and publish?

I know people who have push rights to a repo can publish that repo or install the package for that repo but this means anyone with a github account can publish his/her package to my server. How do I restrict access to a subset of users (ideally users of a github organization). 

For a private enterprise deployment, it is a very very bad default to give everyone access by default.

**Comment by owais** [comment]
@evil-shrike Have you figured out how to restrict access to members of a particular github org?

**Comment by evil-shrike** [comment]
@owais no, I gave up with npme and used https://github.com/hayes/unpm. But now I migrated to Artifactory.

**Comment by owais** [comment]
@evil-shrike Cool! NPME is one of the worst documented projects I've ever come across.

**Comment by soldair** [comment]
hey @evil-shrike we have a new installer and version of npme you should give it a shot if you want.

I know you are aware of this but ill expand.
The default auth plugin uses github and the packages' repository field at time of publish. You can make more auth plugins to leverage internal team and identity management but github/github enterprise is the default out of box.

here is a plain example. 
https://github.com/bcoe/npme-auth-foo
and here is the github plugin
https://github.com/npm/npme-auth-github

I bet you could edit the github plugin to only allow certain users =)

Another thing to think about. npme is a service normally run inside a firewall/vpn to control outside access. npme is not designed to be a firewall. 

On the other hand "private modules" and "Teams and Orgs" (soon to be released) are more along the lines of what you may want if you need people to install/publish your private packages from anywhere internet.

Sorry you got left by the wayside. please email support if you have any other issues especially with npme. It plugs into our ticketing system so issues don't get lost if people move on or get distracted.

hope that helps! =)

**Comment by kevinSuttle** [comment]
So, a few things I'm noticing during the install. 

1. The instructions on the [docs page](https://docs.npmjs.com/enterprise/intro) and the instructions in the CLI after the install are not 1:1. In fact, they're pretty different. 

```bash
Congrats! Your npm on-site server is now up and running \o/

There are just a few final steps:

Step 1. Access your server via HTTPS on port 8800
Step 2. Proceed passed the HTTPS connection security warning (a selfsigned cert is being used initially)
Step 3. Upload a custom TLS/SSL cert/key or proceed with the provided self-signed pair.
Step 4. Configure your npm instance & click "Save".
Step 5. Visit https://docs.npmjs.com/, for information about using npm on-site or contact support@npmjs.com

root@kevin-test:~# npme start
No command 'npme' found, did you mean:
 Command 'npm' from package 'npm' (universe)
npme: command not found
```

2. I was not prompted for [any input](https://docs.npmjs.com/enterprise/installation#installation-questions) during the install, so I can't access my server on port `8800`. I feel like [this config section](https://docs.npmjs.com/enterprise/intro#configuring-the-client) should be listed first. 

3.

**Comment by ashleygwilliams** [comment]
hi @kevinSuttle thanks for filing this!

sadly, we know that the current enterprise docs aren't up to our quality standards. they are definitely on the list, though sadly the list is pretty long. i'm planning on working on them next week, but in the meantime, please feel free to file more issues (the more specific the better!) and/or make a pull request with some updates. i'm happy to walk you through that process if you aren't used to it.

thanks, and sorry, again!

**Comment by kevinSuttle** [comment]
@ashleygwilliams Totally get it. I think the issue here is that there are too many docs. There are at least 4 different sources I see. Getting it down to Web and CLI would be a good start I think, but I'm not in your shoes so I lack context in terms of priorities, etc.

**Comment by soldair** [comment]
this sounds like exactly the kind of thing that is high priority for the onsite team.

**Comment by bcoe** [comment]
@kevinSuttle we've just hired quite a few new people on the on-site team, and plan to start overhauling documentation as one of our first priorities -- getting up and running is much easier than it's been in the past, and trimming down documentation is going to be the brunt of the work.

For the time being, the 5-steps you outlined above:

```Step 1. Access your server via HTTPS on port 8800
Step 2. Proceed passed the HTTPS connection security warning (a selfsigned cert is being used initially)
Step 3. Upload a custom TLS/SSL cert/key or proceed with the provided self-signed pair.
Step 4. Configure your npm instance & click "Save".
Step 5. Visit https://docs.npmjs.com/, for information about using npm on-site or contact support@npmjs.com
```

Should help you get up and running with the admin console. Please email `support@npmjs.com` if you run into any issues, and I'm here to help.

**Comment by kevinSuttle** [comment]
Awesome. Thanks folks. I'll open up specific issues if I run into anything else.

**Comment by kevinSuttle** [comment]
Gah. I just saw yet another differing installation command in the email. Could be why I'm seeing issues.

**Comment by ashleygwilliams** [comment]
hi @kevinSuttle @evil-shrike! we just merged a new getting started guide for npmO, 4b81993410234f5446047bcbbf32e273fc26f82b. it should be live by end of day. closing. please reopen if you continue to have trouble.
